---
title: Exploratory Data Analysis and Normalization
vignette: >
  % \VignetteIndexEntry{EDA}
  % \VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
output:
  html_document:
    mathjax: null   
---

# Setup

```{r setup, message = FALSE, warning=FALSE}
library(SummarizedExperiment)
library(airway)
library(edgeR)
library(EDASeq)
library(ggplot2)
theme_set(theme_bw())
```

# The Airway data

```{r}
data(airway)
airway
colData(airway)
```

# Gene filtering

In this dataset, we have measurements for `r nrow(airway)` genes. Not all of them will be expressed in this particular system.

```{r}
table(rowSums(assay(airway))>0)
```

In general, it is advisable to remove those genes with a low expression level. Often, we may consider a filter that keeps those genes that are expressed in at least some reads in a certain number of samples.

The `filterByExpr` function of the `edgeR` package does that, considering by default 10 reads in at least 10 samples.

```{r}
filter <- filterByExpr(airway)
table(filter)
filtered <- airway[filter,]
filtered
```

# Exploratory Data Analysis

Exploratory Data Analysis (EDA) is useful to highlight those genes that have lower quality or that can have a high leverage in the downstream analyses.

## Total number of reads

First, we need to compute the total number of mapped reads per sample.

```{r}
airway$sum <- colSums(assay(airway))

df <- as.data.frame(colData(airway))

ggplot(df, aes(x=SampleName, y=sum, fill=cell)) +
    geom_bar(stat="identity")

ggplot(df, aes(x=SampleName, y=sum, fill=dex)) +
    geom_bar(stat="identity")
```

While there is clearly a difference in the total number of reads, it does not appear to be related to either cell type or treatment.

## Count distribution

Another immediately available plot is the boxplot of the count distribution.
Given the skewness of count distributions, we typically perform a log transformation.

```{r}
pal <- palette()[-1]
boxplot(log1p(assay(filtered)), col=pal[filtered$cell], las=2)
```

Remember that the `SummarizedExperiment` object may contain multiple matrices (with the same dimension). Hence, we can "save" the log-transformed matrix in the object to avoid computing the log every time that we want to plot their distribution.

```{r}
assay(filtered, "logcounts") <- log1p(assay(filtered))
filtered
```

## Relative Log Expression (RLE)

Sometimes, the log transformation is not enough to highlight issues with the sample distributions and the need for normalization. 

In such cases, we can consider applying the Relative Log Expression (RLE) transformation, defined as
$$
z_{ij} = \log\left(\frac{y_{ij}}{\bar{y}_{\cdot j}}\right) = \log(y_{ij}) - \log(\bar{y}_{\cdot j}),
$$
where $y_{ij}$ is the read count of gene $j$ in sample $i$ and $\bar{y}_{\cdot j}$ is the median expression of gene $j$ across all samples.

The `EDASeq` Bioconductor package contains the `plotRLE` function, which can be simply used in this way.

```{r}
plotRLE(assay(filtered), col=pal[filtered$cell], outline=FALSE, las=2)
```

## Principal Component Analysis (PCA)

To understand which sample characteristics influence gene expression, it is a good idea to observe the first two or three principal components. Ideally, we would like the samples to group by biological variables and not technical noise.

The `plotPCA` function of the `EDASeq` package can be used for this goal. Alternatively, one can use any of the R packages to compute the principal components.

```{r}
tmp <- assay(filtered)
colnames(tmp) <- filtered$cell
plotPCA(tmp, col=pal[filtered$dex])
```

# Normalization

Most of the differences that we see in the RLE plot are not driven by biological signal but due to technical differences between samples, e.g., the total number of reads sequenced for each sample.

We can try to remove such differences via a normalization procedure. There are several different normalization methods proposed for RNA-seq data, but the simplest and often most effective scale each sample by a constant.

The `EDASeq` package implements several such methods, including the "upper-quartile" methods that "aligns" the upper quartile of the distribution of each sample.

Again, we can store the normalized matrix into an assay of the object.

```{r}
assay(filtered, "uq") <- betweenLaneNormalization(assay(filtered), which="upper")
```

We can check with a boxplot that the normalization worked as planned.

```{r}
boxplot(log1p(assay(filtered, "uq")), col=pal[filtered$cell], las=2)
```

Often, the RLE plot is useful to verify that the normalized data look good or to compare normalization methods. We want the median to be as close as possible to zero and the variability to be as balanced as possible between the samples.

```{r}
plotRLE(assay(filtered, "uq"), col=pal[filtered$cell], las=2, outline=FALSE)
```

Another useful diagnostic plot is the plot of the first two principal components. We expect the direction of most variability to be explained by biology rather than technical differences.

```{r}
tmp <- assay(filtered, "uq")
colnames(tmp) <- filtered$cell
plotPCA(tmp, col=pal[filtered$dex])
```

Unlike unnormalized data, after normalization the first principal component separates treated and control samples.

# Session Info

```{r sessionInfo}
sessionInfo()
```
