---
title: Introduction to Bioconductor and its S4 classes
vignette: >
  % \VignetteIndexEntry{Introduction}
  % \VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
output:
  html_document:
    mathjax: null   
---

# Setup

```{r setup, message = FALSE, warning=FALSE}
library(SummarizedExperiment)
library(GenomicRanges)
library(airway)
```

# The `SummarizedExperiment` class

One of the main strengths of the Bioconductor project lies in the use of a common data infrastructure that powers interoperability across packages. 

Users should be able to analyze their data using functions from different Bioconductor packages without the need to convert between formats. To this end, the `SummarizedExperiment` class (from the _SummarizedExperiment_ package) serves as the common currency for data exchange across hundreds of Bioconductor packages.

This class implements a data structure that stores all aspects of the data - gene-by-sample expression data, per-sample metadata and per-gene annotation - and manipulate them in a synchronized manner.

```{r, echo=FALSE}
knitr::include_graphics("https://raw.githubusercontent.com/Bioconductor/SummarizedExperiment/devel/vignettes/SE.svg")
```

Let's start with an example dataset.

```{r, message = FALSE}
data(airway)
airway
```

We can think of this (and other) class as a _container_, that contains several different pieces of data in so-called _slots_.

The _getter_ methods are used to extract information from the slots and the _setter_ methods are used to add information into the slots. These are the only ways to interact with the objects (rather than directly accessing the slots).

Depending on the object, slots can contain different types of data (e.g., numeric matrices, lists, etc.). We will here review the main slots of the SummarizedExperiment class as well as their getter/setter methods.

## The `assays`

This is arguably the most fundamental part of the object that contains the count matrix, and potentially other matrices with transformed data. We can access the _list_ of matrices with the `assays` function and individual matrices with the `assay` function.

```{r}
assay(airway)[1:3, 1:3]
```

You will notice that in this case we have a regular matrix inside the object. More generally, any "matrix-like" object can be used, e.g., sparse matrices or HDF5-backed matrices.

## The `colData` and `rowData`

Conceptually, these are two data frames that annotate the columns and the rows of your assay, respectively.

One can interact with them as usual, e.g., by extracting columns or adding additional variables as columns.

```{r}
colData(airway)
rowData(airway)
```

Note the `$` short cut.

```{r}
identical(colData(airway)$cell, airway$cell)
airway$my_sum <- colSums(assay(airway))
colData(airway)
```

## The `rowRanges`

You might have noticed that our example object is of a special type of SummarizedExperiment class.

```{r}
class(airway)
```

This means that in addition to the already-discussed slots, it contains a `rowRanges` component, wich is a `GRanges` object with the information on the genomic locations of the genes.

```{r}
class(rowRanges(airway))
rowRanges(airway)
```

# GRanges and GRangesList operations

A number of operations are implemented for GRanges, including `subset`, `split`, `length`, `subsetByOverlap` and others.

For instance, we can find the number of exons of the first gene and their average length in nucleotides with the following code.

```{r}
# Exctract the first gene
gene1 <- rowRanges(airway)[[1]]

# Number of exons
length(gene1)

# Average exon length
gene1 |> width() |> mean()
```

# Further reading

[The Bioconductor Project carpentries course](https://carpentries-incubator.github.io/bioc-project/)

# Session Info

```{r sessionInfo}
sessionInfo()
```



