<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exploratory Data Analysis and Normalization • ISCB.Course</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Exploratory Data Analysis and Normalization">
<meta property="og:description" content="ISCB.Course">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ISCB.Course</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/differential_expression.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/eda_norm.html">Exploratory Data Analysis and Normalization</a>
    </li>
    <li>
      <a href="../articles/intro_bioc.html">Introduction to Bioconductor and its S4 classes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drisso/ISCB.Course" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Exploratory Data Analysis and Normalization</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drisso/ISCB.Course/blob/HEAD/vignettes/eda_norm.Rmd" class="external-link"><code>vignettes/eda_norm.Rmd</code></a></small>
      <div class="hidden name"><code>eda_norm.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drisso/EDASeq" class="external-link">EDASeq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>TRADURRE IN INGLESE TUTTO</li>
</ul>
</div>
<div class="section level2">
<h2 id="the-airway-data">The Airway data<a class="anchor" aria-label="anchor" href="#the-airway-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="va">airway</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 63677 8 </span></span>
<span><span class="co">## metadata(1): ''</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(63677): ENSG00000000003 ENSG00000000005 ... ENSG00000273492</span></span>
<span><span class="co">##   ENSG00000273493</span></span>
<span><span class="co">## rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span><span class="co">## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span><span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 8 rows and 9 columns</span></span>
<span><span class="co">##            SampleName     cell      dex    albut        Run avgLength</span></span>
<span><span class="co">##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;</span></span>
<span><span class="co">## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126</span></span>
<span><span class="co">## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126</span></span>
<span><span class="co">## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126</span></span>
<span><span class="co">## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87</span></span>
<span><span class="co">## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120</span></span>
<span><span class="co">## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126</span></span>
<span><span class="co">## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101</span></span>
<span><span class="co">## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98</span></span>
<span><span class="co">##            Experiment    Sample    BioSample</span></span>
<span><span class="co">##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;</span></span>
<span><span class="co">## SRR1039508  SRX384345 SRS508568 SAMN02422669</span></span>
<span><span class="co">## SRR1039509  SRX384346 SRS508567 SAMN02422675</span></span>
<span><span class="co">## SRR1039512  SRX384349 SRS508571 SAMN02422678</span></span>
<span><span class="co">## SRR1039513  SRX384350 SRS508572 SAMN02422670</span></span>
<span><span class="co">## SRR1039516  SRX384353 SRS508575 SAMN02422682</span></span>
<span><span class="co">## SRR1039517  SRX384354 SRS508576 SAMN02422673</span></span>
<span><span class="co">## SRR1039520  SRX384357 SRS508579 SAMN02422683</span></span>
<span><span class="co">## SRR1039521  SRX384358 SRS508580 SAMN02422677</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="gene-filtering">Gene filtering<a class="anchor" aria-label="anchor" href="#gene-filtering"></a>
</h2>
<p>In this dataset, we have measurements for 63677 genes. Not all of
them will be expressed in this particular system.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">## 30208 33469</span></span></code></pre>
<p>In general, it is advisable to remove those genes with a low
expression level. Often, we may consider a filter that keeps those genes
that are expressed in at least some reads in a certain number of
samples.</p>
<p>The <code>filterByExpr</code> function of the <code>edgeR</code>
package does that, considering by default 10 reads in at least 10
samples.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in filterByExpr.DGEList(y, design = design, group = group, lib.size =</span></span>
<span><span class="co">## lib.size, : All samples appear to belong to the same group.</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## filter</span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">## 49453 14224</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtered</span> <span class="op">&lt;-</span> <span class="va">airway</span><span class="op">[</span><span class="va">filter</span>,<span class="op">]</span></span>
<span><span class="va">filtered</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 14224 8 </span></span>
<span><span class="co">## metadata(1): ''</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(14224): ENSG00000000003 ENSG00000000419 ... ENSG00000273382</span></span>
<span><span class="co">##   ENSG00000273486</span></span>
<span><span class="co">## rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span><span class="co">## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span><span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="exploratory-data-analysis">Exploratory Data Analysis<a class="anchor" aria-label="anchor" href="#exploratory-data-analysis"></a>
</h2>
<p>Exploratory Data Analysis (EDA) is useful to highlight those genes
that have lower quality or that can have a high leverage in the
downstream analyses.</p>
<div class="section level3">
<h3 id="total-number-of-reads">Total number of reads<a class="anchor" aria-label="anchor" href="#total-number-of-reads"></a>
</h3>
<p>First, we need to compute the total number of mapped reads per
sample.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">airway</span><span class="op">$</span><span class="va">sum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">SampleName</span>, y<span class="op">=</span><span class="va">sum</span>, fill<span class="op">=</span><span class="va">cell</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">SampleName</span>, y<span class="op">=</span><span class="va">sum</span>, fill<span class="op">=</span><span class="va">dex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-4-2.png" width="700"></p>
<p>While there is clearly a difference in the total number of reads, it
does not appear to be related to either cell type or treatment.</p>
</div>
<div class="section level3">
<h3 id="count-distribution">Count distribution<a class="anchor" aria-label="anchor" href="#count-distribution"></a>
</h3>
<p>Another immediately available plot is the boxplot of the count
distribution. Given the skewness of count distributions, we typically
perform a log transformation.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span><span class="op">)</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">cell</span><span class="op">]</span>, las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>Se vogliamo evitare di calcolare i logaritmi per ogni grafico,
possiamo salvare la matrice dei logaritmi in un nuovo slot dell’oggetto
<code>filtered</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"logcounts"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">filtered</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 14224 8 </span></span>
<span><span class="co">## metadata(1): ''</span></span>
<span><span class="co">## assays(2): counts logcounts</span></span>
<span><span class="co">## rownames(14224): ENSG00000000003 ENSG00000000419 ... ENSG00000273382</span></span>
<span><span class="co">##   ENSG00000273486</span></span>
<span><span class="co">## rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span><span class="co">## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span><span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="relative-log-expression-rle">Relative Log Expression (RLE)<a class="anchor" aria-label="anchor" href="#relative-log-expression-rle"></a>
</h3>
<p>Come detto a lezione, la distribuzione RLE è spesso più utile di
quella dei conteggi, specialmente per confrontare le
normalizzazioni.</p>
<p>Il pacchetto <code>EDASeq</code> contiene la funzione
<code>plotRLE</code> che può essere usata in questo caso.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/plotRLE-methods.html" class="external-link">plotRLE</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">cell</span><span class="op">]</span>, outline<span class="op">=</span><span class="cn">FALSE</span>, las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="analisi-delle-componenti-principali">Analisi delle componenti principali<a class="anchor" aria-label="anchor" href="#analisi-delle-componenti-principali"></a>
</h3>
<p>Osservare come le prime due o tre componenti principali segregano i
campioni può aiutare a capire quali sono le variabili che più
influenzano i valori di espressione. Idealmente, vorremmo che i campioni
si raggruppassero per classe biologica.</p>
<p>Il pacchetto <code>EDASeq</code> ha una funzione <code>plotPCA</code>
che può essere usata in questo caso. In alternativa la PCA può essere
calcolata manualmente con la solita funzione <code><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp()</a></code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">filtered</span><span class="op">$</span><span class="va">cell</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">tmp</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">dex</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="normalization">Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"uq"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/betweenLaneNormalization-methods.html" class="external-link">betweenLaneNormalization</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span>, which<span class="op">=</span><span class="st">"upper"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"uq"</span><span class="op">)</span><span class="op">)</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">cell</span><span class="op">]</span>, las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/plotRLE-methods.html" class="external-link">plotRLE</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"uq"</span><span class="op">)</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">cell</span><span class="op">]</span>, las<span class="op">=</span><span class="fl">2</span>, outline<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"uq"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">filtered</span><span class="op">$</span><span class="va">cell</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">tmp</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">dex</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Davide Risso.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
