<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exploratory Data Analysis and Normalization • ISCB.Course</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Exploratory Data Analysis and Normalization">
<meta property="og:description" content="ISCB.Course">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ISCB.Course</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/differential_expression.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/eda_norm.html">Exploratory Data Analysis and Normalization</a>
    </li>
    <li>
      <a href="../articles/intro_bioc.html">Introduction to Bioconductor and its S4 classes</a>
    </li>
    <li>
      <a href="../articles/multiomics.html">TCGA case study</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drisso/ISCB.Course" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Exploratory Data Analysis and Normalization</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drisso/ISCB.Course/blob/HEAD/vignettes/eda_norm.Rmd" class="external-link"><code>vignettes/eda_norm.Rmd</code></a></small>
      <div class="hidden name"><code>eda_norm.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drisso/EDASeq" class="external-link">EDASeq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-airway-data">The Airway data<a class="anchor" aria-label="anchor" href="#the-airway-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="va">airway</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 63677 8 </span></span>
<span><span class="co">## metadata(1): ''</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(63677): ENSG00000000003 ENSG00000000005 ... ENSG00000273492</span></span>
<span><span class="co">##   ENSG00000273493</span></span>
<span><span class="co">## rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span><span class="co">## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span><span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 8 rows and 9 columns</span></span>
<span><span class="co">##            SampleName     cell      dex    albut        Run avgLength</span></span>
<span><span class="co">##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;</span></span>
<span><span class="co">## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126</span></span>
<span><span class="co">## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126</span></span>
<span><span class="co">## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126</span></span>
<span><span class="co">## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87</span></span>
<span><span class="co">## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120</span></span>
<span><span class="co">## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126</span></span>
<span><span class="co">## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101</span></span>
<span><span class="co">## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98</span></span>
<span><span class="co">##            Experiment    Sample    BioSample</span></span>
<span><span class="co">##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;</span></span>
<span><span class="co">## SRR1039508  SRX384345 SRS508568 SAMN02422669</span></span>
<span><span class="co">## SRR1039509  SRX384346 SRS508567 SAMN02422675</span></span>
<span><span class="co">## SRR1039512  SRX384349 SRS508571 SAMN02422678</span></span>
<span><span class="co">## SRR1039513  SRX384350 SRS508572 SAMN02422670</span></span>
<span><span class="co">## SRR1039516  SRX384353 SRS508575 SAMN02422682</span></span>
<span><span class="co">## SRR1039517  SRX384354 SRS508576 SAMN02422673</span></span>
<span><span class="co">## SRR1039520  SRX384357 SRS508579 SAMN02422683</span></span>
<span><span class="co">## SRR1039521  SRX384358 SRS508580 SAMN02422677</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="gene-filtering">Gene filtering<a class="anchor" aria-label="anchor" href="#gene-filtering"></a>
</h2>
<p>In this dataset, we have measurements for 63677 genes. Not all of
them will be expressed in this particular system.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">## 30208 33469</span></span></code></pre>
<p>In general, it is advisable to remove those genes with a low
expression level. Often, we may consider a filter that keeps those genes
that are expressed in at least some reads in a certain number of
samples.</p>
<p>The <code>filterByExpr</code> function of the <code>edgeR</code>
package does that, considering by default 10 reads in at least 10
samples.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in filterByExpr.DGEList(y, design = design, group = group, lib.size =</span></span>
<span><span class="co">## lib.size, : All samples appear to belong to the same group.</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">filter</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## filter</span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">## 49453 14224</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtered</span> <span class="op">&lt;-</span> <span class="va">airway</span><span class="op">[</span><span class="va">filter</span>,<span class="op">]</span></span>
<span><span class="va">filtered</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 14224 8 </span></span>
<span><span class="co">## metadata(1): ''</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(14224): ENSG00000000003 ENSG00000000419 ... ENSG00000273382</span></span>
<span><span class="co">##   ENSG00000273486</span></span>
<span><span class="co">## rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span><span class="co">## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span><span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="exploratory-data-analysis">Exploratory Data Analysis<a class="anchor" aria-label="anchor" href="#exploratory-data-analysis"></a>
</h2>
<p>Exploratory Data Analysis (EDA) is useful to highlight those genes
that have lower quality or that can have a high leverage in the
downstream analyses.</p>
<div class="section level3">
<h3 id="total-number-of-reads">Total number of reads<a class="anchor" aria-label="anchor" href="#total-number-of-reads"></a>
</h3>
<p>First, we need to compute the total number of mapped reads per
sample.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">airway</span><span class="op">$</span><span class="va">sum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">SampleName</span>, y<span class="op">=</span><span class="va">sum</span>, fill<span class="op">=</span><span class="va">cell</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">SampleName</span>, y<span class="op">=</span><span class="va">sum</span>, fill<span class="op">=</span><span class="va">dex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-4-2.png" width="700"></p>
<p>While there is clearly a difference in the total number of reads, it
does not appear to be related to either cell type or treatment.</p>
</div>
<div class="section level3">
<h3 id="count-distribution">Count distribution<a class="anchor" aria-label="anchor" href="#count-distribution"></a>
</h3>
<p>Another immediately available plot is the boxplot of the count
distribution. Given the skewness of count distributions, we typically
perform a log transformation.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palette.html" class="external-link">palette</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span><span class="op">)</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">cell</span><span class="op">]</span>, las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>Remember that the <code>SummarizedExperiment</code> object may
contain multiple matrices (with the same dimension). Hence, we can
“save” the log-transformed matrix in the object to avoid computing the
log every time that we want to plot their distribution.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"logcounts"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">filtered</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 14224 8 </span></span>
<span><span class="co">## metadata(1): ''</span></span>
<span><span class="co">## assays(2): counts logcounts</span></span>
<span><span class="co">## rownames(14224): ENSG00000000003 ENSG00000000419 ... ENSG00000273382</span></span>
<span><span class="co">##   ENSG00000273486</span></span>
<span><span class="co">## rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span><span class="co">## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span><span class="co">## colData names(9): SampleName cell ... Sample BioSample</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="relative-log-expression-rle">Relative Log Expression (RLE)<a class="anchor" aria-label="anchor" href="#relative-log-expression-rle"></a>
</h3>
<p>Sometimes, the log transformation is not enough to highlight issues
with the sample distributions and the need for normalization.</p>
<p>In such cases, we can consider applying the Relative Log Expression
(RLE) transformation, defined as <span class="math display">\[
z_{ij} = \log\left(\frac{y_{ij}}{\bar{y}_{\cdot j}}\right) =
\log(y_{ij}) - \log(\bar{y}_{\cdot j}),
\]</span> where <span class="math inline">\(y_{ij}\)</span> is the read
count of gene <span class="math inline">\(j\)</span> in sample <span class="math inline">\(i\)</span> and <span class="math inline">\(\bar{y}_{\cdot j}\)</span> is the median
expression of gene <span class="math inline">\(j\)</span> across all
samples.</p>
<p>The <code>EDASeq</code> Bioconductor package contains the
<code>plotRLE</code> function, which can be simply used in this way.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/plotRLE-methods.html" class="external-link">plotRLE</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">cell</span><span class="op">]</span>, outline<span class="op">=</span><span class="cn">FALSE</span>, las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="principal-component-analysis-pca">Principal Component Analysis (PCA)<a class="anchor" aria-label="anchor" href="#principal-component-analysis-pca"></a>
</h3>
<p>To understand which sample characteristics influence gene expression,
it is a good idea to observe the first two or three principal
components. Ideally, we would like the samples to group by biological
variables and not technical noise.</p>
<p>The <code>plotPCA</code> function of the <code>EDASeq</code> package
can be used for this goal. Alternatively, one can use any of the R
packages to compute the principal components.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">filtered</span><span class="op">$</span><span class="va">cell</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">tmp</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">dex</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="normalization">Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<p>Most of the differences that we see in the RLE plot are not driven by
biological signal but due to technical differences between samples,
e.g., the total number of reads sequenced for each sample.</p>
<p>We can try to remove such differences via a normalization procedure.
There are several different normalization methods proposed for RNA-seq
data, but the simplest and often most effective scale each sample by a
constant.</p>
<p>The <code>EDASeq</code> package implements several such methods,
including the “upper-quartile” methods that “aligns” the upper quartile
of the distribution of each sample.</p>
<p>Again, we can store the normalized matrix into an assay of the
object.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"uq"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/betweenLaneNormalization-methods.html" class="external-link">betweenLaneNormalization</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span><span class="op">)</span>, which<span class="op">=</span><span class="st">"upper"</span><span class="op">)</span></span></code></pre></div>
<p>We can check with a boxplot that the normalization worked as
planned.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"uq"</span><span class="op">)</span><span class="op">)</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">cell</span><span class="op">]</span>, las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>Often, the RLE plot is useful to verify that the normalized data look
good or to compare normalization methods. We want the median to be as
close as possible to zero and the variability to be as balanced as
possible between the samples.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/EDASeq/man/plotRLE-methods.html" class="external-link">plotRLE</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"uq"</span><span class="op">)</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">cell</span><span class="op">]</span>, las<span class="op">=</span><span class="fl">2</span>, outline<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>Another useful diagnostic plot is the plot of the first two principal
components. We expect the direction of most variability to be explained
by biology rather than technical differences.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">filtered</span>, <span class="st">"uq"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">filtered</span><span class="op">$</span><span class="va">cell</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">tmp</span>, col<span class="op">=</span><span class="va">pal</span><span class="op">[</span><span class="va">filtered</span><span class="op">$</span><span class="va">dex</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="eda_norm_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>Unlike unnormalized data, after normalization the first principal
component separates treated and control samples.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ggplot2_3.4.3               EDASeq_2.34.0              </span></span>
<span><span class="co">##  [3] ShortRead_1.58.0            GenomicAlignments_1.36.0   </span></span>
<span><span class="co">##  [5] Rsamtools_2.16.0            Biostrings_2.68.1          </span></span>
<span><span class="co">##  [7] XVector_0.40.0              BiocParallel_1.34.2        </span></span>
<span><span class="co">##  [9] edgeR_3.42.4                limma_3.56.2               </span></span>
<span><span class="co">## [11] airway_1.20.0               SummarizedExperiment_1.30.2</span></span>
<span><span class="co">## [13] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">## [15] GenomeInfoDb_1.36.1         IRanges_2.34.1             </span></span>
<span><span class="co">## [17] S4Vectors_0.38.1            BiocGenerics_0.46.0        </span></span>
<span><span class="co">## [19] MatrixGenerics_1.12.3       matrixStats_1.0.0          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] DBI_1.1.3               bitops_1.0-7            deldir_1.0-9           </span></span>
<span><span class="co">##  [4] biomaRt_2.56.1          rlang_1.1.1             magrittr_2.0.3         </span></span>
<span><span class="co">##  [7] compiler_4.3.0          RSQLite_2.3.1           GenomicFeatures_1.52.1 </span></span>
<span><span class="co">## [10] png_0.1-8               systemfonts_1.0.4       vctrs_0.6.3            </span></span>
<span><span class="co">## [13] stringr_1.5.0           pkgconfig_2.0.3         crayon_1.5.2           </span></span>
<span><span class="co">## [16] fastmap_1.1.1           dbplyr_2.3.3            labeling_0.4.2         </span></span>
<span><span class="co">## [19] utf8_1.2.3              rmarkdown_2.24          ragg_1.2.5             </span></span>
<span><span class="co">## [22] purrr_1.0.2             bit_4.0.5               xfun_0.40              </span></span>
<span><span class="co">## [25] zlibbioc_1.46.0         cachem_1.0.8            jsonlite_1.8.7         </span></span>
<span><span class="co">## [28] progress_1.2.2          blob_1.2.4              highr_0.10             </span></span>
<span><span class="co">## [31] DelayedArray_0.26.7     jpeg_0.1-10             parallel_4.3.0         </span></span>
<span><span class="co">## [34] prettyunits_1.1.1       R6_2.5.1                bslib_0.5.1            </span></span>
<span><span class="co">## [37] stringi_1.7.12          RColorBrewer_1.1-3      rtracklayer_1.60.1     </span></span>
<span><span class="co">## [40] jquerylib_0.1.4         Rcpp_1.0.11             knitr_1.43             </span></span>
<span><span class="co">## [43] R.utils_2.12.2          Matrix_1.6-1            tidyselect_1.2.0       </span></span>
<span><span class="co">## [46] abind_1.4-5             yaml_2.3.7              codetools_0.2-19       </span></span>
<span><span class="co">## [49] hwriter_1.3.2.1         curl_5.0.2              lattice_0.21-8         </span></span>
<span><span class="co">## [52] tibble_3.2.1            withr_2.5.0             KEGGREST_1.40.0        </span></span>
<span><span class="co">## [55] evaluate_0.21           desc_1.4.2              BiocFileCache_2.8.0    </span></span>
<span><span class="co">## [58] xml2_1.3.5              pillar_1.9.0            filelock_1.0.2         </span></span>
<span><span class="co">## [61] generics_0.1.3          rprojroot_2.0.3         RCurl_1.98-1.12        </span></span>
<span><span class="co">## [64] hms_1.1.3               munsell_0.5.0           scales_1.2.1           </span></span>
<span><span class="co">## [67] glue_1.6.2              tools_4.3.0             interp_1.1-4           </span></span>
<span><span class="co">## [70] BiocIO_1.10.0           locfit_1.5-9.8          fs_1.6.3               </span></span>
<span><span class="co">## [73] XML_3.99-0.14           grid_4.3.0              latticeExtra_0.6-30    </span></span>
<span><span class="co">## [76] colorspace_2.1-0        AnnotationDbi_1.62.2    GenomeInfoDbData_1.2.10</span></span>
<span><span class="co">## [79] restfulr_0.0.15         cli_3.6.1               rappdirs_0.3.3         </span></span>
<span><span class="co">## [82] textshaping_0.3.6       fansi_1.0.4             S4Arrays_1.0.5         </span></span>
<span><span class="co">## [85] dplyr_1.1.2             gtable_0.3.4            R.methodsS3_1.8.2      </span></span>
<span><span class="co">## [88] sass_0.4.7              digest_0.6.33           aroma.light_3.30.0     </span></span>
<span><span class="co">## [91] farver_2.1.1            rjson_0.2.21            memoise_2.0.1          </span></span>
<span><span class="co">## [94] htmltools_0.5.6         pkgdown_2.0.7           R.oo_1.25.0            </span></span>
<span><span class="co">## [97] lifecycle_1.0.3         httr_1.4.7              bit64_4.0.5</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Davide Risso.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
