<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>TCGA case study • ISCB.Course</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="TCGA case study">
<meta property="og:description" content="ISCB.Course">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ISCB.Course</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/differential_expression.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/eda_norm.html">Exploratory Data Analysis and Normalization</a>
    </li>
    <li>
      <a href="../articles/intro_bioc.html">Introduction to Bioconductor and its S4 classes</a>
    </li>
    <li>
      <a href="../articles/multiomics.html">TCGA case study</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drisso/ISCB.Course" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>TCGA case study</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/drisso/ISCB.Course/blob/HEAD/vignettes/multiomics.Rmd" class="external-link"><code>vignettes/multiomics.Rmd</code></a></small>
      <div class="hidden name"><code>multiomics.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">curatedTCGAData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TCGAutils</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/therneau/survival" class="external-link">survival</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/survminer/index.html" class="external-link">survminer</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-multiassayexperiment-class">The MultiAssayExperiment class<a class="anchor" aria-label="anchor" href="#the-multiassayexperiment-class"></a>
</h2>
<p><img src="https://waldronlab.io/MultiAssayWorkshop/articles/MultiAssayExperiment.png"><!-- --></p>
<p>In modern “multi-omic” experiments, several different omic data are
collected on the same samples, e.g., RNA-seq, mutations, copy-number
variation, etc.</p>
<p>In these cases, it is useful to organize the data in a single object
that has the ability of linking together samples from the same
individuals and data types of different nature.</p>
<p>The Bioconductor class for this task is the
<code>MultiAssayExperiment</code>. Conceptually, it can be thought of as
a list of objects, one per data modality, e.g., a SummarizedExperiment
for the transcritpomic profiles, but it contains several features that
make working with multi-omic data simpler.</p>
<p>In order to see it in action, we look at an example from The Cancer
Genome Atlas (TCGA).</p>
</div>
<div class="section level2">
<h2 id="curated-tcga-data">Curated TCGA data<a class="anchor" aria-label="anchor" href="#curated-tcga-data"></a>
</h2>
<p>Bioconductor includes the <code>curatedTCGAData</code> package, which
contains a curated version of the multi-omic TCGA dataset. These data
consist of thousands of samples across different cancer types.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">diseaseCodes</span><span class="op">)</span></span>
<span><span class="va">diseaseCodes</span></span></code></pre></div>
<pre><code><span><span class="co">##    Study.Abbreviation Available SubtypeData</span></span>
<span><span class="co">## 1                 ACC       Yes         Yes</span></span>
<span><span class="co">## 2                BLCA       Yes         Yes</span></span>
<span><span class="co">## 3                BRCA       Yes         Yes</span></span>
<span><span class="co">## 4                CESC       Yes          No</span></span>
<span><span class="co">## 5                CHOL       Yes          No</span></span>
<span><span class="co">## 6                CNTL        No          No</span></span>
<span><span class="co">## 7                COAD       Yes         Yes</span></span>
<span><span class="co">## 8                DLBC       Yes          No</span></span>
<span><span class="co">## 9                ESCA       Yes          No</span></span>
<span><span class="co">## 10               FPPP        No          No</span></span>
<span><span class="co">## 11                GBM       Yes         Yes</span></span>
<span><span class="co">## 12               HNSC       Yes         Yes</span></span>
<span><span class="co">## 13               KICH       Yes         Yes</span></span>
<span><span class="co">## 14               KIRC       Yes         Yes</span></span>
<span><span class="co">## 15               KIRP       Yes         Yes</span></span>
<span><span class="co">## 16               LAML       Yes         Yes</span></span>
<span><span class="co">## 17               LCML        No          No</span></span>
<span><span class="co">## 18                LGG       Yes         Yes</span></span>
<span><span class="co">## 19               LIHC       Yes          No</span></span>
<span><span class="co">## 20               LUAD       Yes         Yes</span></span>
<span><span class="co">## 21               LUSC       Yes         Yes</span></span>
<span><span class="co">## 22               MESO       Yes          No</span></span>
<span><span class="co">## 23               MISC        No          No</span></span>
<span><span class="co">## 24                 OV       Yes         Yes</span></span>
<span><span class="co">## 25               PAAD       Yes          No</span></span>
<span><span class="co">## 26               PCPG       Yes          No</span></span>
<span><span class="co">## 27               PRAD       Yes         Yes</span></span>
<span><span class="co">## 28               READ       Yes          No</span></span>
<span><span class="co">## 29               SARC       Yes          No</span></span>
<span><span class="co">## 30               SKCM       Yes         Yes</span></span>
<span><span class="co">## 31               STAD       Yes         Yes</span></span>
<span><span class="co">## 32               TGCT       Yes          No</span></span>
<span><span class="co">## 33               THCA       Yes         Yes</span></span>
<span><span class="co">## 34               THYM       Yes          No</span></span>
<span><span class="co">## 35               UCEC       Yes         Yes</span></span>
<span><span class="co">## 36                UCS       Yes          No</span></span>
<span><span class="co">## 37                UVM       Yes          No</span></span>
<span><span class="co">##                                                          Study.Name</span></span>
<span><span class="co">## 1                                          Adrenocortical carcinoma</span></span>
<span><span class="co">## 2                                      Bladder Urothelial Carcinoma</span></span>
<span><span class="co">## 3                                         Breast invasive carcinoma</span></span>
<span><span class="co">## 4  Cervical squamous cell carcinoma and endocervical adenocarcinoma</span></span>
<span><span class="co">## 5                                                Cholangiocarcinoma</span></span>
<span><span class="co">## 6                                                          Controls</span></span>
<span><span class="co">## 7                                              Colon adenocarcinoma</span></span>
<span><span class="co">## 8                   Lymphoid Neoplasm Diffuse Large B-cell Lymphoma</span></span>
<span><span class="co">## 9                                              Esophageal carcinoma</span></span>
<span><span class="co">## 10                                              FFPE Pilot Phase II</span></span>
<span><span class="co">## 11                                          Glioblastoma multiforme</span></span>
<span><span class="co">## 12                            Head and Neck squamous cell carcinoma</span></span>
<span><span class="co">## 13                                               Kidney Chromophobe</span></span>
<span><span class="co">## 14                                Kidney renal clear cell carcinoma</span></span>
<span><span class="co">## 15                            Kidney renal papillary cell carcinoma</span></span>
<span><span class="co">## 16                                           Acute Myeloid Leukemia</span></span>
<span><span class="co">## 17                                     Chronic Myelogenous Leukemia</span></span>
<span><span class="co">## 18                                         Brain Lower Grade Glioma</span></span>
<span><span class="co">## 19                                   Liver hepatocellular carcinoma</span></span>
<span><span class="co">## 20                                              Lung adenocarcinoma</span></span>
<span><span class="co">## 21                                     Lung squamous cell carcinoma</span></span>
<span><span class="co">## 22                                                     Mesothelioma</span></span>
<span><span class="co">## 23                                                    Miscellaneous</span></span>
<span><span class="co">## 24                                Ovarian serous cystadenocarcinoma</span></span>
<span><span class="co">## 25                                        Pancreatic adenocarcinoma</span></span>
<span><span class="co">## 26                               Pheochromocytoma and Paraganglioma</span></span>
<span><span class="co">## 27                                          Prostate adenocarcinoma</span></span>
<span><span class="co">## 28                                            Rectum adenocarcinoma</span></span>
<span><span class="co">## 29                                                          Sarcoma</span></span>
<span><span class="co">## 30                                          Skin Cutaneous Melanoma</span></span>
<span><span class="co">## 31                                           Stomach adenocarcinoma</span></span>
<span><span class="co">## 32                                      Testicular Germ Cell Tumors</span></span>
<span><span class="co">## 33                                                Thyroid carcinoma</span></span>
<span><span class="co">## 34                                                          Thymoma</span></span>
<span><span class="co">## 35                             Uterine Corpus Endometrial Carcinoma</span></span>
<span><span class="co">## 36                                           Uterine Carcinosarcoma</span></span>
<span><span class="co">## 37                                                   Uveal Melanoma</span></span></code></pre>
<p>Let’s focus on the Adrenocortical carcinoma (ACC) and let’s see what
type of data are available for this cohort.</p>
<p>Adrenocortical carcinoma is a rare disease in which malignant
(cancer) cells form in the outer layer of the adrenal gland.</p>
<p><img src="https://nci-media.cancer.gov/pdq/media/images/739009-750.jpg"><!-- --></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html" class="external-link">curatedTCGAData</a></span><span class="op">(</span><span class="st">"ACC"</span>, version <span class="op">=</span> <span class="st">"2.0.1"</span>, dry.run <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## See '?curatedTCGAData' for 'diseaseCode' and 'assays' inputs</span></span></code></pre>
<pre><code><span><span class="co">##     ah_id                                 title file_size</span></span>
<span><span class="co">## 1  EH4737                   ACC_CNASNP-20160128    0.8 Mb</span></span>
<span><span class="co">## 2  EH4738                   ACC_CNVSNP-20160128    0.2 Mb</span></span>
<span><span class="co">## 3  EH4740         ACC_GISTIC_AllByGene-20160128    0.2 Mb</span></span>
<span><span class="co">## 4  EH4741             ACC_GISTIC_Peaks-20160128      0 Mb</span></span>
<span><span class="co">## 5  EH4742 ACC_GISTIC_ThresholdedByGene-20160128    0.2 Mb</span></span>
<span><span class="co">## 6  EH4744       ACC_Methylation-20160128_assays  239.2 Mb</span></span>
<span><span class="co">## 7  EH4745           ACC_Methylation-20160128_se      6 Mb</span></span>
<span><span class="co">## 8  EH4746             ACC_miRNASeqGene-20160128    0.1 Mb</span></span>
<span><span class="co">## 9  EH4747                 ACC_Mutation-20160128    0.7 Mb</span></span>
<span><span class="co">## 10 EH4748              ACC_RNASeq2Gene-20160128    2.7 Mb</span></span>
<span><span class="co">## 11 EH4749          ACC_RNASeq2GeneNorm-20160128      4 Mb</span></span>
<span><span class="co">## 12 EH4750                ACC_RPPAArray-20160128    0.1 Mb</span></span>
<span><span class="co">##                    rdataclass rdatadateadded rdatadateremoved</span></span>
<span><span class="co">## 1            RaggedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 2            RaggedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 3        SummarizedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 4  RangedSummarizedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 5        SummarizedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 6        SummarizedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 7            RaggedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 8        SummarizedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 9        SummarizedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 10       SummarizedExperiment     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 11                     DFrame     2021-01-27             &lt;NA&gt;</span></span>
<span><span class="co">## 12       SummarizedExperiment     2021-01-27             &lt;NA&gt;</span></span></code></pre>
<p>We can for instance download gene expression, micro-RNA expression,
mutations, and copy-number variation data from this cohort.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html" class="external-link">curatedTCGAData</a></span><span class="op">(</span></span>
<span>    diseaseCode <span class="op">=</span> <span class="st">"ACC"</span>,</span>
<span>    assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"miRNASeqGene"</span>, <span class="st">"Mutation"</span>, <span class="st">"RNASeq2GeneNorm"</span>, <span class="st">"CNVSNP"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    version <span class="op">=</span> <span class="st">"2.0.1"</span>,</span>
<span>    dry.run <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Working on: ACC_CNVSNP-20160128</span></span></code></pre>
<pre><code><span><span class="co">## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## require("RaggedExperiment")</span></span></code></pre>
<pre><code><span><span class="co">## Working on: ACC_miRNASeqGene-20160128</span></span></code></pre>
<pre><code><span><span class="co">## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Working on: ACC_Mutation-20160128</span></span></code></pre>
<pre><code><span><span class="co">## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Working on: ACC_RNASeq2GeneNorm-20160128</span></span></code></pre>
<pre><code><span><span class="co">## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Working on: ACC_colData-20160128</span></span></code></pre>
<pre><code><span><span class="co">## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Working on: ACC_metadata-20160128</span></span></code></pre>
<pre><code><span><span class="co">## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Working on: ACC_sampleMap-20160128</span></span></code></pre>
<pre><code><span><span class="co">## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 655 sampleMap rows not in names(experiments)</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">acc</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] ACC_CNVSNP-20160128: RaggedExperiment with 21052 rows and 180 columns</span></span>
<span><span class="co">##  [2] ACC_miRNASeqGene-20160128: SummarizedExperiment with 1046 rows and 80 columns</span></span>
<span><span class="co">##  [3] ACC_Mutation-20160128: RaggedExperiment with 20166 rows and 90 columns</span></span>
<span><span class="co">##  [4] ACC_RNASeq2GeneNorm-20160128: SummarizedExperiment with 20501 rows and 79 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>There are several utility functions that can be used to navigate this
complex object:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html" class="external-link">experiments</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] ACC_CNVSNP-20160128: RaggedExperiment with 21052 rows and 180 columns</span></span>
<span><span class="co">##  [2] ACC_miRNASeqGene-20160128: SummarizedExperiment with 1046 rows and 80 columns</span></span>
<span><span class="co">##  [3] ACC_Mutation-20160128: RaggedExperiment with 20166 rows and 90 columns</span></span>
<span><span class="co">##  [4] ACC_RNASeq2GeneNorm-20160128: SummarizedExperiment with 20501 rows and 79 columns</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html" class="external-link">sampleMap</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 429 rows and 3 columns</span></span>
<span><span class="co">##                            assay      primary                colname</span></span>
<span><span class="co">##                         &lt;factor&gt;  &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">## 1   ACC_RNASeq2GeneNorm-20160128 TCGA-OR-A5J1 TCGA-OR-A5J1-01A-11R..</span></span>
<span><span class="co">## 2   ACC_RNASeq2GeneNorm-20160128 TCGA-OR-A5J2 TCGA-OR-A5J2-01A-11R..</span></span>
<span><span class="co">## 3   ACC_RNASeq2GeneNorm-20160128 TCGA-OR-A5J3 TCGA-OR-A5J3-01A-11R..</span></span>
<span><span class="co">## 4   ACC_RNASeq2GeneNorm-20160128 TCGA-OR-A5J5 TCGA-OR-A5J5-01A-11R..</span></span>
<span><span class="co">## 5   ACC_RNASeq2GeneNorm-20160128 TCGA-OR-A5J6 TCGA-OR-A5J6-01A-31R..</span></span>
<span><span class="co">## ...                          ...          ...                    ...</span></span>
<span><span class="co">## 425        ACC_Mutation-20160128 TCGA-PK-A5H8 TCGA-PK-A5H8-01A-11D..</span></span>
<span><span class="co">## 426        ACC_Mutation-20160128 TCGA-PK-A5H9 TCGA-PK-A5H9-01A-11D..</span></span>
<span><span class="co">## 427        ACC_Mutation-20160128 TCGA-PK-A5HA TCGA-PK-A5HA-01A-11D..</span></span>
<span><span class="co">## 428        ACC_Mutation-20160128 TCGA-PK-A5HB TCGA-PK-A5HB-01A-11D..</span></span>
<span><span class="co">## 429        ACC_Mutation-20160128 TCGA-PK-A5HC TCGA-PK-A5HC-01A-11D..</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html" class="external-link">sampleTables</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $`ACC_CNVSNP-20160128`</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 01 10 11 </span></span>
<span><span class="co">## 90 85  5 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`ACC_miRNASeqGene-20160128`</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 01 </span></span>
<span><span class="co">## 80 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`ACC_Mutation-20160128`</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 01 </span></span>
<span><span class="co">## 90 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`ACC_RNASeq2GeneNorm-20160128`</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 01 </span></span>
<span><span class="co">## 79</span></span></code></pre>
<p>In particular, the <code>sampleTables</code> function tells us which
type of samples we have in our object. We can use the following code for
a human-readable version of the codes.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sampleTypes"</span><span class="op">)</span></span>
<span><span class="va">sampleTypes</span></span></code></pre></div>
<pre><code><span><span class="co">##    Code                                        Definition Short.Letter.Code</span></span>
<span><span class="co">## 1    01                               Primary Solid Tumor                TP</span></span>
<span><span class="co">## 2    02                             Recurrent Solid Tumor                TR</span></span>
<span><span class="co">## 3    03   Primary Blood Derived Cancer - Peripheral Blood                TB</span></span>
<span><span class="co">## 4    04      Recurrent Blood Derived Cancer - Bone Marrow              TRBM</span></span>
<span><span class="co">## 5    05                          Additional - New Primary               TAP</span></span>
<span><span class="co">## 6    06                                        Metastatic                TM</span></span>
<span><span class="co">## 7    07                             Additional Metastatic               TAM</span></span>
<span><span class="co">## 8    08                        Human Tumor Original Cells              THOC</span></span>
<span><span class="co">## 9    09        Primary Blood Derived Cancer - Bone Marrow               TBM</span></span>
<span><span class="co">## 10   10                              Blood Derived Normal                NB</span></span>
<span><span class="co">## 11   11                               Solid Tissue Normal                NT</span></span>
<span><span class="co">## 12   12                                Buccal Cell Normal               NBC</span></span>
<span><span class="co">## 13   13                           EBV Immortalized Normal              NEBV</span></span>
<span><span class="co">## 14   14                                Bone Marrow Normal               NBM</span></span>
<span><span class="co">## 15   15                                    sample type 15              15SH</span></span>
<span><span class="co">## 16   16                                    sample type 16              16SH</span></span>
<span><span class="co">## 17   20                                   Control Analyte             CELLC</span></span>
<span><span class="co">## 18   40 Recurrent Blood Derived Cancer - Peripheral Blood               TRB</span></span>
<span><span class="co">## 19   50                                        Cell Lines              CELL</span></span>
<span><span class="co">## 20   60                          Primary Xenograft Tissue                XP</span></span>
<span><span class="co">## 21   61                Cell Line Derived Xenograft Tissue               XCL</span></span>
<span><span class="co">## 22   99                                    sample type 99              99SH</span></span></code></pre>
<p>In addition to the omic data, we also have clinical and demographic
information for each patient.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html" class="external-link">getSubtypeMap</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          ACC_annotations     ACC_subtype</span></span>
<span><span class="co">## 1             Patient_ID       patientID</span></span>
<span><span class="co">## 2  histological_subtypes       Histology</span></span>
<span><span class="co">## 3          mrna_subtypes         C1A/C1B</span></span>
<span><span class="co">## 4          mrna_subtypes         mRNA_K4</span></span>
<span><span class="co">## 5                   cimp      MethyLevel</span></span>
<span><span class="co">## 6      microrna_subtypes   miRNA cluster</span></span>
<span><span class="co">## 7          scna_subtypes    SCNA cluster</span></span>
<span><span class="co">## 8       protein_subtypes protein cluster</span></span>
<span><span class="co">## 9   integrative_subtypes             COC</span></span>
<span><span class="co">## 10     mutation_subtypes        OncoSign</span></span></code></pre>
<p>For instance, we can see that the tumors have been classified in
terms of their histology and that using mutation data clinicians have
determined several molecular subtypes.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span><span class="op">$</span><span class="va">Histology</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##      Myxoid Type   Oncocytic Type Sarcomatoid Type       Usual Type </span></span>
<span><span class="co">##                1                4                2               84</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span><span class="op">$</span><span class="va">OncoSign</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##          CN1          CN2       CTNNB1   TERT/ZNRF3     TP53/NF1 Unclassified </span></span>
<span><span class="co">##           10           14            6           17           24           16</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="primary-tumor-rna-seq">Primary tumor RNA-seq<a class="anchor" aria-label="anchor" href="#primary-tumor-rna-seq"></a>
</h2>
<p>Let’s focus on primary tumor samples and let’s extract the RNA-seq
data.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">primaryTumors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/TCGAprimaryTumors.html" class="external-link">TCGAprimaryTumors</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 90 sampleMap rows with 'colname' not in colnames of experiments</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html" class="external-link">sampleTables</a></span><span class="op">(</span><span class="va">primaryTumors</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $`ACC_CNVSNP-20160128`</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 01 </span></span>
<span><span class="co">## 90 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`ACC_miRNASeqGene-20160128`</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 01 </span></span>
<span><span class="co">## 80 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`ACC_Mutation-20160128`</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 01 </span></span>
<span><span class="co">## 90 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`ACC_RNASeq2GeneNorm-20160128`</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 01 </span></span>
<span><span class="co">## 79</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment-helpers.html" class="external-link">getWithColData</a></span><span class="op">(</span><span class="va">primaryTumors</span>, <span class="st">"ACC_RNASeq2GeneNorm-20160128"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 20501 79 </span></span>
<span><span class="co">## metadata(3): filename build platform</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(20501): A1BG A1CF ... psiTPTE22 tAKR</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(79): TCGA-OR-A5J1 TCGA-OR-A5J2 ... TCGA-PK-A5HA TCGA-PK-A5HB</span></span>
<span><span class="co">## colData names(822): patientID years_to_birth ... genome_doublings ADS</span></span></code></pre>
<p>As you can see we have hundreds of colData columns, but only a
minority of them are useful for the analysis. We can use the following
code to focus on the clinically-relevant features.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html" class="external-link">getClinicalNames</a></span><span class="op">(</span><span class="st">"ACC"</span><span class="op">)</span></span>
<span><span class="va">cl</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "years_to_birth"                      </span></span>
<span><span class="co">##  [2] "vital_status"                        </span></span>
<span><span class="co">##  [3] "days_to_death"                       </span></span>
<span><span class="co">##  [4] "days_to_last_followup"               </span></span>
<span><span class="co">##  [5] "tumor_tissue_site"                   </span></span>
<span><span class="co">##  [6] "pathologic_stage"                    </span></span>
<span><span class="co">##  [7] "pathology_T_stage"                   </span></span>
<span><span class="co">##  [8] "pathology_N_stage"                   </span></span>
<span><span class="co">##  [9] "gender"                              </span></span>
<span><span class="co">## [10] "date_of_initial_pathologic_diagnosis"</span></span>
<span><span class="co">## [11] "radiation_therapy"                   </span></span>
<span><span class="co">## [12] "histological_type"                   </span></span>
<span><span class="co">## [13] "residual_tumor"                      </span></span>
<span><span class="co">## [14] "number_of_lymph_nodes"               </span></span>
<span><span class="co">## [15] "race"                                </span></span>
<span><span class="co">## [16] "ethnicity"</span></span></code></pre>
<p>We can get simple exploratory statistics on them to get an idea of
the characteristics of the cohort.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">vital_status</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  0  1 </span></span>
<span><span class="co">## 51 28</span></span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">pathologic_stage</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##   stage i  stage ii stage iii  stage iv </span></span>
<span><span class="co">##         9        37        16        15</span></span></code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">gender</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## female   male </span></span>
<span><span class="co">##     48     31</span></span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">radiation_therapy</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  no yes </span></span>
<span><span class="co">##  59  17</span></span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">years_to_birth</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##    14.0    35.0    49.0    46.7    59.5    77.0</span></span></code></pre>
<div class="section level3">
<h3 id="differential-expression">Differential expression<a class="anchor" aria-label="anchor" href="#differential-expression"></a>
</h3>
<p>Obviously, everything that we have done in the previous vignette can
be done on these data as well. Here, some care is required on deciding
which covariates to include in the model, as the usual issues with
confounders in observational studies apply.</p>
<p>Here, we decide to add gender and age as confounders as we test the
difference in gene expression between molecular subtypes.</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">se</span><span class="op">[</span>,<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">OncoSign</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">se</span>,</span>
<span>                    design <span class="op">=</span> <span class="op">~</span> <span class="va">gender</span> <span class="op">+</span> <span class="va">years_to_birth</span> <span class="op">+</span> <span class="va">OncoSign</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## renaming the first element in assays to 'counts'</span></span></code></pre>
<pre><code><span><span class="co">## converting counts to integer mode</span></span></code></pre>
<pre><code><span><span class="co">## Warning in DESeqDataSet(se, design = ~gender + years_to_birth + OncoSign): some</span></span>
<span><span class="co">## variables in design formula are characters, converting to factors</span></span></code></pre>
<pre><code><span><span class="co">##   the design formula contains one or more numeric variables with integer values,</span></span>
<span><span class="co">##   specifying a model with increasing fold change for higher values.</span></span>
<span><span class="co">##   did you mean for this to be a factor? if so, first convert</span></span>
<span><span class="co">##   this variable to a factor using the factor() function</span></span></code></pre>
<pre><code><span><span class="co">##   the design formula contains one or more numeric variables that have mean or</span></span>
<span><span class="co">##   standard deviation larger than 5 (an arbitrary threshold to trigger this message).</span></span>
<span><span class="co">##   Including numeric variables with large mean can induce collinearity with the intercept.</span></span>
<span><span class="co">##   Users should center and scale numeric variables in the design to improve GLM convergence.</span></span></code></pre>
<pre><code><span><span class="co">##   Note: levels of factors in the design contain characters other than</span></span>
<span><span class="co">##   letters, numbers, '_' and '.'. It is recommended (but not required) to use</span></span>
<span><span class="co">##   only letters, numbers, and delimiters '_' or '.', as these are safe characters</span></span>
<span><span class="co">##   for column names in R. [This is a message, not a warning or an error]</span></span></code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## estimating size factors</span></span>
<span><span class="co">##   Note: levels of factors in the design contain characters other than</span></span>
<span><span class="co">##   letters, numbers, '_' and '.'. It is recommended (but not required) to use</span></span>
<span><span class="co">##   only letters, numbers, and delimiters '_' or '.', as these are safe characters</span></span>
<span><span class="co">##   for column names in R. [This is a message, not a warning or an error]</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">##   Note: levels of factors in the design contain characters other than</span></span>
<span><span class="co">##   letters, numbers, '_' and '.'. It is recommended (but not required) to use</span></span>
<span><span class="co">##   only letters, numbers, and delimiters '_' or '.', as these are safe characters</span></span>
<span><span class="co">##   for column names in R. [This is a message, not a warning or an error]</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<pre><code><span><span class="co">## 34 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest</span></span></code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span>, contrast <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"OncoSign"</span>, <span class="st">"TERT/ZNRF3"</span>, <span class="st">"TP53/NF1"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## out of 19600 with nonzero total read count</span></span>
<span><span class="co">## adjusted p-value &lt; 0.1</span></span>
<span><span class="co">## LFC &gt; 0 (up)       : 36, 0.18%</span></span>
<span><span class="co">## LFC &lt; 0 (down)     : 336, 1.7%</span></span>
<span><span class="co">## outliers [1]       : 0, 0%</span></span>
<span><span class="co">## low counts [2]     : 1896, 9.7%</span></span>
<span><span class="co">## (mean count &lt; 1)</span></span>
<span><span class="co">## [1] see 'cooksCutoff' argument of ?results</span></span>
<span><span class="co">## [2] see 'independentFiltering' argument of ?results</span></span></code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## log2 fold change (MLE): OncoSign TERT/ZNRF3 vs TP53/NF1 </span></span>
<span><span class="co">## Wald test p-value: OncoSign TERT/ZNRF3 vs TP53/NF1 </span></span>
<span><span class="co">## DataFrame with 6 rows and 6 columns</span></span>
<span><span class="co">##            baseMean log2FoldChange     lfcSE      stat      pvalue        padj</span></span>
<span><span class="co">##           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">## CALML5     0.275942       32.99735  4.167824   7.91716 2.42989e-15          NA</span></span>
<span><span class="co">## ANPEP    392.268477       -4.52832  0.690949  -6.55377 5.61006e-11 9.93204e-07</span></span>
<span><span class="co">## MPP3      57.496064       -2.87737  0.458757  -6.27210 3.56216e-10 3.15323e-06</span></span>
<span><span class="co">## USH1C     46.847855       -7.55418  1.228636  -6.14843 7.82528e-10 4.53297e-06</span></span>
<span><span class="co">## DKK1      97.187249       -4.17584  0.683937  -6.10560 1.02417e-09 4.53297e-06</span></span>
<span><span class="co">## PRAMEF18   0.575532      -25.28543  4.164355  -6.07187 1.26428e-09          NA</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="survival-analysis">Survival Analysis<a class="anchor" aria-label="anchor" href="#survival-analysis"></a>
</h2>
<p>Survival analysis of different disease subtypes is obviously
relevant.</p>
<p>Here we look at overal survival using a standard model and look at
the Kaplan-Meier curves contrasting two molecular subtypes.</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span><span class="op">$</span><span class="va">overall_survival</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">vital_status</span> <span class="op">==</span> <span class="fl">1</span>,</span>
<span>                                   <span class="va">se</span><span class="op">$</span><span class="va">days_to_death</span>,</span>
<span>                                   <span class="va">se</span><span class="op">$</span><span class="va">days_to_last_followup</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_sub</span> <span class="op">&lt;-</span> <span class="va">se</span><span class="op">[</span>,<span class="va">se</span><span class="op">$</span><span class="va">OncoSign</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TERT/ZNRF3"</span>, <span class="st">"TP53/NF1"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">sfit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survfit.html" class="external-link">survfit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html" class="external-link">Surv</a></span><span class="op">(</span><span class="va">overall_survival</span>, <span class="va">vital_status</span><span class="op">)</span><span class="op">~</span><span class="va">OncoSign</span>,</span>
<span>                data<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se_sub</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html" class="external-link">ggsurvplot</a></span><span class="op">(</span><span class="va">sfit</span>, conf.int<span class="op">=</span><span class="cn">TRUE</span>, pval <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="multiomics_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>Here, we see evidence that the two subtypes have different overall
survival, although more sophisticated survival models are likely
needed.</p>
<p>A typical workflow, which we do not have time to work through, is to
use differential expression to identify “interesting” genes and then
look at a classifier, often called a <em>signature</em> in this context,
able to group patients based on their survival. The usual care of
machine learning approaches (e.g., cross-validation and
train-test-validation splits) is required as in other cases.</p>
</div>
<div class="section level2">
<h2 id="further-reading">Further reading<a class="anchor" aria-label="anchor" href="#further-reading"></a>
</h2>
<p><a href="https://waldronlab.io/MultiAssayWorkshop" class="external-link">The Waldron Lab
MultiAssayWorkshop</a></p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] RaggedExperiment_1.24.2     survminer_0.4.9            </span></span>
<span><span class="co">##  [3] ggpubr_0.6.0                ggplot2_3.4.3              </span></span>
<span><span class="co">##  [5] survival_3.5-7              DESeq2_1.40.2              </span></span>
<span><span class="co">##  [7] TCGAutils_1.20.2            curatedTCGAData_1.22.2     </span></span>
<span><span class="co">##  [9] MultiAssayExperiment_1.26.0 SummarizedExperiment_1.30.2</span></span>
<span><span class="co">## [11] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">## [13] GenomeInfoDb_1.36.1         IRanges_2.34.1             </span></span>
<span><span class="co">## [15] S4Vectors_0.38.1            BiocGenerics_0.46.0        </span></span>
<span><span class="co">## [17] MatrixGenerics_1.12.3       matrixStats_1.0.0          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] jsonlite_1.8.7                magrittr_2.0.3               </span></span>
<span><span class="co">##   [3] GenomicFeatures_1.52.1        farver_2.1.1                 </span></span>
<span><span class="co">##   [5] rmarkdown_2.24                fs_1.6.3                     </span></span>
<span><span class="co">##   [7] BiocIO_1.10.0                 zlibbioc_1.46.0              </span></span>
<span><span class="co">##   [9] ragg_1.2.5                    vctrs_0.6.3                  </span></span>
<span><span class="co">##  [11] memoise_2.0.1                 Rsamtools_2.16.0             </span></span>
<span><span class="co">##  [13] RCurl_1.98-1.12               rstatix_0.7.2                </span></span>
<span><span class="co">##  [15] BiocBaseUtils_1.2.0           htmltools_0.5.6              </span></span>
<span><span class="co">##  [17] S4Arrays_1.0.5                progress_1.2.2               </span></span>
<span><span class="co">##  [19] AnnotationHub_3.8.0           curl_5.0.2                   </span></span>
<span><span class="co">##  [21] broom_1.0.5                   sass_0.4.7                   </span></span>
<span><span class="co">##  [23] bslib_0.5.1                   desc_1.4.2                   </span></span>
<span><span class="co">##  [25] zoo_1.8-12                    cachem_1.0.8                 </span></span>
<span><span class="co">##  [27] GenomicAlignments_1.36.0      mime_0.12                    </span></span>
<span><span class="co">##  [29] lifecycle_1.0.3               pkgconfig_2.0.3              </span></span>
<span><span class="co">##  [31] Matrix_1.6-1                  R6_2.5.1                     </span></span>
<span><span class="co">##  [33] fastmap_1.1.1                 GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">##  [35] shiny_1.7.5                   digest_0.6.33                </span></span>
<span><span class="co">##  [37] colorspace_2.1-0              AnnotationDbi_1.62.2         </span></span>
<span><span class="co">##  [39] rprojroot_2.0.3               ExperimentHub_2.8.1          </span></span>
<span><span class="co">##  [41] textshaping_0.3.6             RSQLite_2.3.1                </span></span>
<span><span class="co">##  [43] labeling_0.4.2                filelock_1.0.2               </span></span>
<span><span class="co">##  [45] km.ci_0.5-6                   fansi_1.0.4                  </span></span>
<span><span class="co">##  [47] httr_1.4.7                    abind_1.4-5                  </span></span>
<span><span class="co">##  [49] compiler_4.3.0                bit64_4.0.5                  </span></span>
<span><span class="co">##  [51] withr_2.5.0                   backports_1.4.1              </span></span>
<span><span class="co">##  [53] BiocParallel_1.34.2           carData_3.0-5                </span></span>
<span><span class="co">##  [55] DBI_1.1.3                     highr_0.10                   </span></span>
<span><span class="co">##  [57] ggsignif_0.6.4                biomaRt_2.56.1               </span></span>
<span><span class="co">##  [59] rappdirs_0.3.3                DelayedArray_0.26.7          </span></span>
<span><span class="co">##  [61] rjson_0.2.21                  tools_4.3.0                  </span></span>
<span><span class="co">##  [63] interactiveDisplayBase_1.38.0 httpuv_1.6.11                </span></span>
<span><span class="co">##  [65] glue_1.6.2                    restfulr_0.0.15              </span></span>
<span><span class="co">##  [67] promises_1.2.1                grid_4.3.0                   </span></span>
<span><span class="co">##  [69] generics_0.1.3                gtable_0.3.4                 </span></span>
<span><span class="co">##  [71] KMsurv_0.1-5                  tzdb_0.4.0                   </span></span>
<span><span class="co">##  [73] tidyr_1.3.0                   data.table_1.14.8            </span></span>
<span><span class="co">##  [75] hms_1.1.3                     xml2_1.3.5                   </span></span>
<span><span class="co">##  [77] car_3.1-2                     utf8_1.2.3                   </span></span>
<span><span class="co">##  [79] XVector_0.40.0                BiocVersion_3.17.1           </span></span>
<span><span class="co">##  [81] pillar_1.9.0                  stringr_1.5.0                </span></span>
<span><span class="co">##  [83] later_1.3.1                   splines_4.3.0                </span></span>
<span><span class="co">##  [85] dplyr_1.1.2                   BiocFileCache_2.8.0          </span></span>
<span><span class="co">##  [87] lattice_0.21-8                rtracklayer_1.60.1           </span></span>
<span><span class="co">##  [89] bit_4.0.5                     tidyselect_1.2.0             </span></span>
<span><span class="co">##  [91] locfit_1.5-9.8                Biostrings_2.68.1            </span></span>
<span><span class="co">##  [93] knitr_1.43                    gridExtra_2.3                </span></span>
<span><span class="co">##  [95] xfun_0.40                     stringi_1.7.12               </span></span>
<span><span class="co">##  [97] yaml_2.3.7                    evaluate_0.21                </span></span>
<span><span class="co">##  [99] codetools_0.2-19              tibble_3.2.1                 </span></span>
<span><span class="co">## [101] BiocManager_1.30.22           cli_3.6.1                    </span></span>
<span><span class="co">## [103] xtable_1.8-4                  systemfonts_1.0.4            </span></span>
<span><span class="co">## [105] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">## [107] survMisc_0.5.6                Rcpp_1.0.11                  </span></span>
<span><span class="co">## [109] GenomicDataCommons_1.24.2     dbplyr_2.3.3                 </span></span>
<span><span class="co">## [111] png_0.1-8                     XML_3.99-0.14                </span></span>
<span><span class="co">## [113] parallel_4.3.0                ellipsis_0.3.2               </span></span>
<span><span class="co">## [115] pkgdown_2.0.7                 readr_2.1.4                  </span></span>
<span><span class="co">## [117] blob_1.2.4                    prettyunits_1.1.1            </span></span>
<span><span class="co">## [119] bitops_1.0-7                  scales_1.2.1                 </span></span>
<span><span class="co">## [121] purrr_1.0.2                   crayon_1.5.2                 </span></span>
<span><span class="co">## [123] rlang_1.1.1                   KEGGREST_1.40.0              </span></span>
<span><span class="co">## [125] rvest_1.0.3</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Davide Risso.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
